src/java.base/share/classes/jdk/internal/classfile/impl/DirectCodeBuilder.java

```diff
@@ -318,52 +369,33 @@ public void writeBody(BufWriter b) {
                             methodInfo.methodName().stringValue(),
                             methodInfo.methodTypeSymbol().displayDescriptor()));
                 }
-                int maxStack, maxLocals;
-                Attribute<? extends StackMapTableAttribute> stackMapAttr;
-                boolean canReuseStackmaps = codeAndExceptionsMatch(codeLength);
 
-                if (!constantPool.options().generateStackmaps) {
-                    StackCounter cntr = StackCounter.of(DirectCodeBuilder.this, buf);
-                    maxStack = cntr.maxStack();
-                    maxLocals = cntr.maxLocals();
-                    stackMapAttr = null;
-                }
-                else if (canReuseStackmaps) {
-                    maxLocals = original.maxLocals();
-                    maxStack = original.maxStack();
-                    stackMapAttr = original.findAttribute(Attributes.STACK_MAP_TABLE).orElse(null);
```

Comment:
On the baseline, if a stack map table attribute can be reused but this findAttribute doesn't find an attribute (implying an implicit stack map, potentially for simple code?), we will not perform a stack map generation and simply reuse a trivial stack map (null), which should be a correct algorithm.

Reply:
Yes, I caused this regression by "Revert of DirectCodeBuilder.needsStackMap pre-calculation" commit above.
The needStackMap pre-calculation on the other side caused regression in lambda performance, so one regression has been replaced with another.